plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -387,8 +390,13 @@ public Map<String, ColumnHandle> getColumnHandles(ConnectorSession session, Conn
     {
         IcebergTableHandle table = (IcebergTableHandle) tableHandle;
         Table icebergTable = catalog.loadTable(session, table.getSchemaTableName());
-        return getColumns(icebergTable.schema(), typeManager).stream()
-                .collect(toImmutableMap(IcebergColumnHandle::getName, identity()));
+
+        ImmutableMap.Builder<String, ColumnHandle> columnHandles = ImmutableMap.builder();
+        for (IcebergColumnHandle columnHandle : getColumns(icebergTable.schema(), typeManager)) {
+            columnHandles.put(columnHandle.getName(), columnHandle);
+        }
+        columnHandles.put(FILE_PATH.columnName(), pathColumnHandle());
+        return columnHandles.build();
```

Comment:
this was the earlier state, and suggested/discussed with @alexjo2144 to use map instead of iterating columns multiple times.
I feel this approach should be ok.

Reply:
please refer: https://github.com/trinodb/trino/pull/11661#discussion_r847650330